cli/reactjs_jsx_ppx_v3.ml

```diff
@@ -604,12 +597,10 @@ let jsxMapper () =
     [@@raises Invalid_argument]
   in
 
-  let argToConcreteType types (name, loc, type_) =
+  let argToConcreteType types (name, _loc, type_) =
     match name with
     | name when isLabelled name -> (false, getLabel name, [], type_) :: types
-    | name when isOptional name ->
-      (true, getLabel name, [], Typ.constr ~loc {loc; txt = optionIdent} [type_])
-      :: types
```

Comment:
> A small comment on style: this looks a little strange:
> 
> ```ocaml
> let makePropsTypeParamsSig namedTypeList =
>   namedTypeList
>   |> List.filter_map (fun (isOptional, label, _, interiorType) ->
>          if label = "key" || label = "ref" then None
>          else if isOptional then Some (extractOptionalCoreType interiorType)
>          else Some interiorType)
> ```
> 
> as it removes an `@optional ` from the type. But that annotation was added by the ppx itself.
> 
> It might indicate that the logic for doing this should probably be further back in the caller, not in this function. So that the `@optional` attribute is never added in the first place. Otherwise, it could be easy to introduce bugs via refactoring without realizing it.

Fixed. I totally forgot this improvised logic at that time. Thanks!